### ISSUE TITLE
Add in priors + fixed weights for LFs

### ISSUE THREAD

**Issue by ajratner** [HazyResearch/snorkel#627]
Internal representation passed to generative model will now be, in addition to labeling function matrix `L` (M x N):
- `priors`: An M-dim vector specifying the prior _probabilities_ of all the LFs
- `is_fixed`: An M-dim sparse vector specifying if the LF weight is fixed (e.g. for hand-labeled data LF) or not

If `prior` is not == 0.5, then a fixed factor will be added to re-center the weight for that LF (which will still be regularized to zero, otherwise annoying NS changes needed)

This will cover (a) passing labeled data, and (b) setting individual priors of the LFs, e.g. based on the dev set accuracies (see separate issue!)

Front end input will be something like:
```
gen_model.train(L_train, LF_priors=[0.5,...,0.5], labeled_data=None, labeled_prior=1.5)
```

TODO list:
- [x] Implementation of new fixed factors in Snorkel
- [x] Test of the above
- [x] Implementation in front end
- [x] Remove redundant `y`
- [x] Change LF initial acc to at prior val
- [x] Implement proper calculation of LF accs. with deps
- [x] Write a few additional sanity-check level tests

**Comment by ajratner** [comment]
Merged into dev
